### ISSUE TITLE
PodPresets can't be enumerated by `kubectl get podpreset`

### ISSUE THREAD

**Issue by IronhandedLayman** [kubernetes/kubernetes#44736]
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.): No

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.): podpresets, kubectl

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Kubernetes version** (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:33:11Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:22:08Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}

**Environment**:
- **Cloud provider or hardware configuration**: Ubuntu 16.10 (Yakkety Yak) 7-node single master Kubernetes cluster
- **OS** (e.g. from /etc/os-release): Client: MacOS Darwin 10.12.4, Server: Ubuntu 16.10 
- **Kernel** (e.g. `uname -a`): 4.8.0.46-generic
- **Install tools**: kubeadm
- **Others**: Using `--feature-gates=AllAlpha=true` on all k8s manifests and kubelet.

**What happened**:
Created a `PodPreset` using the following manifest:
```
kind: PodPreset
apiVersion: settings.k8s.io/v1alpha1
metadata:
  name: busybox-settings
  namespace: default
spec:
  selector:
    matchLabels:
      role: busybox
  env:
    - name: NATE
      value: "isgood"
```
`kubectl create -f busybox-podpreset.yaml` returns `podpreset "busybox-settings" created`, but `kubectl get podpreset` gives me the following error:

```
NAME      KIND
error: unknown type &settings.PodPreset{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"busybox-settings", GenerateName:"", Namespace:"default", SelfLink:"/apis/settings.k8s.io/v1alpha1/namespaces/default/podpresets/busybox-settings", UID:"48fd7c1c-25f7-11e7-87ae-0025905ca872", ResourceVersion:"1574921", Generation:1, CreationTimestamp:v1.Time{Time:time.Time{sec:63628309766, nsec:0, loc:(*time.Location)(0x26c11e0)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:settings.PodPresetSpec{Selector:v1.LabelSelector{MatchLabels:map[string]string{"role":"busybox"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}, Env:[]api.EnvVar{api.EnvVar{Name:"NATE", Value:"isgood", ValueFrom:(*api.EnvVarSource)(nil)}}, EnvFrom:[]api.EnvFromSource(nil), Volumes:[]api.Volume(nil), VolumeMounts:[]api.VolumeMount(nil)}}
``` 

However, `kubectl get podpreset -o yaml busybox-settings` returns the expected manifest:
```
apiVersion: settings.k8s.io/v1alpha1
kind: PodPreset
metadata:
  creationTimestamp: 2017-04-20T18:29:26Z
  generation: 1
  name: busybox-settings
  namespace: default
  resourceVersion: "1574921"
  selfLink: /apis/settings.k8s.io/v1alpha1/namespaces/default/podpresets/busybox-settings
  uid: 48fd7c1c-25f7-11e7-87ae-0025905ca872
spec:
  env:
  - name: NATE
    value: isgood
  selector:
    matchLabels:
      role: busybox
```
Also, the PodPreset works as expected; pods with the label `role: busybox` are receiving the environment variable as indicated in the PodPreset, as well as the following line in the manifest for the pod:
```
metadata:
  annotations:
    podpreset.admission.kubernetes.io/busybox-settings: "1574921"
```

**What you expected to happen**:
I expected `kubectl get podpreset` to return a table of existing PodPresets along with their selectors.

**How to reproduce it** (as minimally and precisely as possible):
As indicated above:
1. Enable a Kubernetes 1.6.2 cluster to use PodPresets
2. Create a PodPreset using the manifest above
3. Run `kubectl get podpreset`

**Anything else we need to know**:
Also of note: `kubectl delete` works fine on PodPresets, either by using the manifest file creating it or the name of the manifest. Also, tab completion works on PodPresets on the command line, as well as `-o json` and `-o jsonpath`. Only `kubectl get podpreset` and `kubectl get podpreset -o wide` seem to be broken.

**Comment by xiangpengzhao** [comment]
yes, `kubectl get podpreset` is not supported yet. But what we could/should display by `kubectl get podpreset` might be only `NAME` and `SELECTOR`.

**Comment by xiangpengzhao** [comment]
Fix this issue in PR #44746.
